Fix static create methods

cristianoc72 opened the pull request:
Fix `phootwork\lang\Text:create()` and `phootwork\file\FileOprationTrait::create()`
to return the correct class, when extend them.

Example:
```php
<?php
use phootwork\file\File;

class FileExtension extends File {}

$fileExt = FileExtension::create('a_file.txt');

var_dump($fileExt instanceof FileExtension::class); 
// false without this commit
//since `create` method returns a `phootwork\file\File` instance
```

gossi reviewed (approved):
Hmm... we had this once? Somehow went gone. But very to have tests to cover this feature. Great work 👍

cristianoc72 commented:
Yes, we had it. I removed it when I fixed some Psalm issues, but I were totally wrong!! :smile: 
I'm gonna write other tests to verify the other methods (I'm using this library intensively at the moment).
